src/main/java/com/atlassian/bitbucket/jenkins/internal/deployments/DeployedToEnvironmentNotifierStep.java

```diff
@@ -0,0 +1,197 @@
+
+    private final BitbucketDeploymentEnvironment environment;
+
+    @DataBoundConstructor
+    public DeployedToEnvironmentNotifierStep(String environmentKey, String environmentName,
+                                             @CheckForNull String environmentType,
+                                             @CheckForNull String environmentUrl) {
+        environment = new BitbucketDeploymentEnvironment.Builder(getOrGenerateEnvironmentKey(environmentKey),
+                environmentName)
+                .type(BitbucketDeploymentEnvironmentType.fromName(stripToNull(environmentType)).orElse(null))
+                .url(environmentUrl)
+                .build();
+    }
+
+    public static DescriptorImpl descriptor() {
+        return Jenkins.get().getDescriptorByType(DescriptorImpl.class);
+    }
+
+    public String getEnvironmentKey() {
+        return environment.getKey();
+    }
+
+    public String getEnvironmentName() {
+        return environment.getName();
+    }
+
+    @CheckForNull
+    public String getEnvironmentType() {
+        return environment.getType();
+    }
+
+    @CheckForNull
+    public String getEnvironmentUrl() {
+        return environment.getUrl();
+    }
+
+    @Override
+    public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher,
+                        TaskListener listener) throws InterruptedException, IOException {
+        try {
+            BitbucketRevisionAction revisionAction = run.getAction(BitbucketRevisionAction.class);
+            if (revisionAction == null) {
+                // Not checked out with a Bitbucket SCM
+                listener.error("Could not send deployment notification: DeployedToEnvironmentNotifierStep only works when using the Bitbucket SCM for checkout.");
+                return;
+            }
+
+            // We use the default call to the bitbucketDeploymentFactory to get the state of the deployment based
+            // on the run.
+            BitbucketDeployment deployment = descriptor().bitbucketDeploymentFactory.createDeployment(run, environment);
+
+            BitbucketSCMRepository bitbucketSCMRepo = revisionAction.getBitbucketSCMRepo();
+            String revisionSha = revisionAction.getRevisionSha1();
+            descriptor().deploymentPoster.postDeployment(bitbucketSCMRepo.getServerId(),
+                    bitbucketSCMRepo.getProjectKey(), bitbucketSCMRepo.getRepositorySlug(), revisionSha, deployment,
+                    run, listener);
+        } catch (RuntimeException e) {
+            // This shouldn't happen because deploymentPoster.postDeployment doesn't throw anything. But just in case,
+            // we don't want to throw anything and potentially stop other steps from being executed
+            String errorMsg =
+                    String.format("An exception occurred when trying to post the deployment to Bitbucket Server: %s", e.getMessage());
```

Comment:
```suggestion
                    String.format("An error occurred when trying to post the deployment to Bitbucket Server: %s", e.getMessage());
```

Our users may not be really familiar with `Exceptions` I guess most people are, but `error` conveys the same thing and being  less language specific.